versions/3.0.md

```diff
@@ -1123,6 +1125,344 @@ application/json:
+
+As opposed to _dynamic_ links--that is, links provided **in** the response payload, the OAS linking mechanism does not require that link information be provided in a specific response format at runtime.
+
+For computing links, and providing instructions to execute them, a mechanism is defined for accessing values in a response and using them as variables while invoking the linked operation.
+
+Field Name | Type | Description
+---|:---:|---
+<a name="referenceRef"></a>$ref | `string` | If present, a reference to another Links Object.  Note, the presence of `$ref` in the Links Object will cause all _Patterned Objects_ to be ignored. 
+
+Field Pattern | Type | Description
+---|:---:|---
+<a name="linkName"></a>link name | [Link Object](#linkObject) | A short name for the link, following the naming constraints of the <a href="#definitionsName">definitions name</a>.  The link shall reference a single Link Object, or a JSON Reference to a single link object
+
+
+### Link Object
+The `Link Object` is responsible for defining a possible operation based on a single response.
+
+Field Name  |  Type  | Description
+  ---       | :---:  |     ---
+href      |  url   | a relative or absolute URL to a linked resource. This field is mutually exclusive with the `operationId` field.
+ operationId| string | the name of an _existing_, resolvable OAS operation, as defined with a unique `operationId`.  This field is mutually exclusive with the `href` field.  Relative `href` values _may_ be used to locate an existing Operation Object in the OAS.
+parameters  | Link Parameters Object | an Object representing parameters to pass to an operation as specified with `operationId` or identified via `href`.
+description | string | a description of the link, supports GFM.
+
+#### <a name="responsePayload"></a>Response Payload Values
+
+Payload values are only available in parseable response payloads which match the advertised media-type.  In all cases, if a value does _not_ exist, the parameter will be considered a `null` value (as opposed to an empty value) and _not_ passed as a parameter to the linked resource. In cases where a value is required, and a parameter is not supplied, the client _may_ choose to not follow the link definition. 
+
+### Example
+
+Response payload:
+```json
+{
+    "id": "df71a505-07bc-458a-a5c0-73c0340d1ec7",
+    "firstname": "Ash",
+    "lastname": "Williams"
+}
+```
+
+Payload Variables:
+```yaml
+id: df71a505-07bc-458a-a5c0-73c0340d1ec7
+firstname: Ash
+lastname: Williams
+missingValue: null
+```
+
+In situations where variables appear in an array, an array of variables will be extracted.  For example:
+
+```json
+[
+    { "color": "red" },
+    { "color": "green" },
+    { "color": "blue" }
+]
+```
+
+will be extracted as such:
+
+```json
+color: ["red", "green", "blue"]
+```
```

Comment:
Another thing I noted while trying to write down my proposal:

* In the `href` field we have an URL template in which parts within `{}` will be interpreted as some kind of expression and then replaced.

* In the link parameters object, we have "static or dynamic values". The specification text doesn't really say what this is, but from the examples it looks like "dynamic values" are strings starting with `$response` (which are interpreted as the same kind of expressions as before), while static values are everything else (i.e. booleans, numbers, objects, arrays, or strings not starting with `$response`). I guess only primitives are sensible for simple parameters (header/query/path), but for body parameters having an object or array would make sense. When using the dynamic values, we could also extract objects/arrays from the response, but then it is not clear how this would interact with the array expansion logic.